server/mailet/rate-limiter-redis/src/main/scala/org/apache/james/rate/limiter/redis/RedisHealthCheck.scala

```diff
@@ -0,0 +1,52 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ * http://www.apache.org/licenses/LICENSE-2.0                   *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.rate.limiter.redis
+
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
+import eu.timepit.refined.auto._
+import javax.inject.Inject
+import org.apache.james.core.healthcheck.{ComponentName, HealthCheck, Result}
+import org.apache.james.rate.limiter.api.{RateLimitingKey, Rule, Rules}
+import org.reactivestreams.Publisher
+import reactor.core.scala.publisher.SMono
+
+import scala.concurrent.duration.FiniteDuration
+
+sealed class RedisHealthCheckKey() extends RateLimitingKey {
+  override def asString(): String = "RedisHealthCheck"
+}
+
+class RedisHealthCheck @Inject()(redisRateLimiterFactory: RedisRateLimiterFactory) extends HealthCheck {
```

Comment:
IMO basing a "generic" component like a healthcheck on an applicative feature is brittle. 

In a redis health check I'd expect only the Redis layer to be tested.